### PR TITLE
samples: Add tictactoe

### PR DESCRIPTION
**Pull request by juliaogris**
Add a text based and a convas/mouse version of TicTacToe.
Allow for `"unlisted"` samples.

Links:
https://evy-lang--135-hik6n21b.web.app#tictactoetext
https://evy-lang--135-hik6n21b.web.app#tictactoe

### CODE REVIEW

**Review by camh- — COMMENTED** [review]
Some comments so far on the first commit. I assume the code of the second commit might also get updated the same? (not sure, havent looked yet).

I did play a few games of the web version on strength 3 and I won each time I started. I would always start in the middle and bot always went somewhere that meant I could win. I know it is quite slow without that - I wonder what else could be done, because at strength 3, it should never lose.

File: frontend/samples/games/tictactoetext.evy [hunk 1]
```diff
@@ -0,0 +1,222 @@
...
+            b2[i] = "x"
+            if (winner b2) == "x"
+                b[i] = "x"
+                return
+            end
+            b2[i] = "."
+        end
+    end
+    // try to stop "o" from winning in next move
+    for i := range 9
```
[the remark this answers is not in the document]
Comment (camh-) [reply frontend/samples/games/tictactoetext.evy:137]: oh. sorry. just realised this is wrong. ignore it please

File: frontend/samples/games/tictactoetext.evy [hunk 2]
```diff
@@ -0,0 +1,222 @@
...
+    print b[3] b[4] b[5]
+    print b[6] b[7] b[8]
+    print
+end
+
+func human_turn b:[]string
+    while true
+        str := read
+        n := str2num str
+        if !err and n <= 8 and n >= 0 and b[n] == "."
```
[the remark this answers is not in the document]
Comment (juliaogris) [reply frontend/samples/games/tictactoetext.evy:91]: ok.

[the remark this answers is not in the document]
Comment (juliaogris) [reply frontend/samples/games/tictactoetext.evy:91]: I played it on the web and had the printboard function right next to it, that made it easier.

File: frontend/samples/games/tictactoetext.evy [hunk 3]
```diff
@@ -0,0 +1,222 @@
...
+            cnt = cnt + 1
+        end
+    end
+end
+
+func bot2_turn b:[]string
+    // try to win in next move
+    for i := range 9
+        if b[i] == "."
+            b2 := b[:]
```
[the remark this answers is not in the document]
Comment (juliaogris) [reply frontend/samples/games/tictactoetext.evy:127]: removed, thank you.

File: frontend/samples/games/tictactoetext.evy (hunk 1 above)
[the remark this answers is not in the document]
Comment (juliaogris) [reply frontend/samples/games/tictactoetext.evy:137]: yeah, i thought about this too. if i bot can win in a later move it should win, right?

**Review by camh- — APPROVED** [review]
:shamrock:

**Pull request merged**